src/HttpRequest.cc

```diff
@@ -738,3 +738,31 @@ HttpRequest::manager(const CbcPointer<ConnStateData> &aMgr, const AccessLogEntry
     }
 }
 
+const Ip::Address *
+FindListeningPortAddress(const HttpRequest *request, const bool derive_ips, const Ip::Address *cacheip, const Ip::Address *localip)
```

Comment:
I chose cacheIp because its is either al->cache.port->s OR request->masterXaction->squidPort->s (squid is a cache server). Hence the name cacheIp.

localIp is either al->tcpClient->local OR request->masterXaction->tcpClient->local ... word local is common to both cases. Hence name localIp.

I could not find any other suitable name.